zulip/zulip/__init__.py

```diff
@@ -1071,6 +1071,24 @@ def get_user_by_id(self, user_id, request=None):
             request=request,
         )
 
+    def update_user_by_id(self, user_id, user_data):
+        # type: (int, Dict[str, Any]) -> Dict[str, Any]
+
+        '''
+            Example usage:
+
+            >>> client.update_user_by_id(8, {'full_name': "NewName"})
+            {'result': 'success', 'msg': ''}
+        '''
+        for key, value in user_data.items():
+            user_data[key] = json.dumps(value)
+
+        return self.call_endpoint(
+            url='users/{}'.format(user_id),
+            method='PATCH',
+            request=user_data
```

Comment:
Why not do `**request` so that the user can just use `client.update_user_by_id(8, full_name="NewName")`?  The `get_streams` endpoint uses this pattern, and I think it's a good one.  

(We should change your other new endpoints, like `get_user_by_id`, to use this as well, while they're still new and we can freely change the Python API)